frontend/catalyst/compiler.py

```diff
@@ -16,14 +16,46 @@
 """
 
 import os
+import sys
 import shutil
 import subprocess
 import warnings
+from io import TextIOWrapper
+from typing import Optional, List
+from dataclasses import dataclass
 
 from catalyst._configuration import INSTALLED
 
 package_root = os.path.dirname(__file__)
 
+
+@dataclass
+class CompileOptions:
+    """Common compile options"""
+
+    verbosity: int
```

Comment:
It could be a boolean, right?
```suggestion
    verbosity: bool
```

Reply:
I agree with @grwlf here, `int` is my preferred choice. We could have different verbose levels. At the moment, it is only one, so a `bool` applies, but in the future this could be expanded.

Reply:
@maliasadi I started right from `int` because often people want more than two levels of verbosity, as Erick mentioned. But it is true that we might not need it right now. Let me know if you really want to see bool.

Reply:
No doubt on the points! But it currently doesn't have any use cases in the scope of this work. Besides, it will be an easy change if one wants to add different verbosity levels at some point in future. I may suggest using `bool` now and this can be updated it anytime in future.

Reply:
Done